giraph-core/src/main/java/org/apache/giraph/master/BspServiceMaster.java

```diff
@@ -984,6 +984,9 @@ private GlobalStats aggregateWorkerStats(long superstep) {
       } else {
         printAggregatedMetricsToHDFS(superstep, aggregatedMetrics);
       }
+      for (MasterObserver observer : observers) {
+        observer.superstepMetricsUpdate(superstep, aggregatedMetrics);
```

Comment:
Should we catch exceptions here?

Reply:
Not sure, we don't do it on other actions of MasterObserver, implementation can always add safety inside of it.